appliances/MinIO/appliance.sh

```diff
@@ -0,0 +1,396 @@
+# ------------------------------------------------------------------------------
+
+ONEAPP_MINIO_ROOT_USER="${ONEAPP_MINIO_ROOT_USER:-myminioadmin}"
+ONEAPP_MINIO_ROOT_PASSWORD="${ONEAPP_MINIO_ROOT_PASSWORD:-minio-secret-key-change-me}"
+ONEAPP_MINIO_OPTS="${ONEAPP_MINIO_OPTS:---console-address :9001}"
+ONEAPP_MINIO_HOSTNAME="${ONEAPP_MINIO_HOSTNAME:-localhost,minio-*.example.net}"
+
+### Globals ##########################################################
+
+MINIO_VERSION="https://dl.min.io/server/minio/release/linux-amd64/archive/minio_20240510014138.0.0_amd64.deb"
+CERTGEN_VERSION="https://github.com/minio/certgen/releases/download/v1.2.1/certgen_1.2.1_linux_amd64.deb"
+
+###############################################################################
+###############################################################################
+###############################################################################
+
+#
+# service implementation
+#
+
+service_cleanup()
+{
+    :
+}
+
+service_install()
+{
+    # ensuring that the setup directory exists
+    #TODO: move to service
+    mkdir -p "$ONE_SERVICE_SETUP_DIR"
+
+    # minio install
+    install_minio_server
+    # certgen install
+    install_minio_certgen
+    # create user and group for MinIO
+    create_user_group
+    # create MinIO environment variable file
+    create_default_environment_variable_file
+    # create MinIO systemd service file
+    create_enable_systemd_service
+    # add MinIO section to fstab
+    msg info "Add MinIO section to fstab"
+    cat >> /etc/fstab <<EOF
+# Start MinIO-volumes
+# End MinIO-volumes
+EOF
+
+    # service metadata
+    create_one_service_metadata
+
+    # cleanup
+    postinstall_cleanup
+
+    msg info "INSTALLATION FINISHED"
+
+    return 0
+}
+
+service_configure()
+{
+    # For each drive other than /dev/vda, check if they're already formatted.
+    # If there's no format, create partitions and filesystem
+    msg info "Checking drives partition and format"
+    local_minio_drives=$(lsblk -d -o NAME | grep vd | grep -v 'vda')
+    local_drives_count=1
+    local_minio_volumes=""
+    for device in $local_minio_drives; do
+        # Partition and format dev
+        partition_format_drive $device
+        # Create folder in /mnt for drive mount
+        folder="/mnt/disk-${local_drives_count}"
+        local_minio_volumes+=" ${folder}"
+
+        if [[ ! -d "${folder}" ]]; then
+            msg info "Create folder ${folder}"
+            mkdir "${folder}"
+            msg info "Add /dev/${device}1 entry to fstab"
+            sed -i "/# End MinIO-volumes/i /dev/${device}1    ${folder}    xfs    defaults    0    0" /etc/fstab
+        else
+            msg info "Folder ${folder} exists, skipping"
+        fi
+
+        local_drives_count=$((local_drives_count+1))
+    done
+    local_drives_count=$((local_drives_count-1))
+
+    msg info "Mount fstab"
+    if ! mount -a -t xfs; then
+        msg error "Error mounting MinIO drives"
```

Comment:
Isn't the fact that you can't mount xfs partitions a fatal error really? If we don't exit here continue using root filesystem how does this impact users, what happens when nobody looks at logs? :thinking:

Reply:
If this fails, the MinIO server won't work, so even if users don't check logs, they will notice since the service won't be running. But we could actually just exit here, since if this fails there the server is doomed